state/peerAccount.go

```diff
@@ -108,9 +108,10 @@ func (pa *peerAccount) SetTempRating(rating uint32) {
 }
 
 // SetListAndIndex will update the peer's list (eligible, waiting) and the index inside it with journal
-func (pa *peerAccount) SetListAndIndex(shardID uint32, list string, index uint32, updatePreviousList bool) {
-	if updatePreviousList {
+func (pa *peerAccount) SetListAndIndex(shardID uint32, list string, index uint32, updatePreviousValues bool) {
+	if updatePreviousValues {
 		pa.PreviousList = pa.List
+		pa.PreviousIndexInList = pa.IndexInList
```

Comment:
We don't need also a variable to hold PreviousShardID ?

Reply:
Apparently, every time SetListAndIndex (jailed/leaving) is called, it uses the same shardID, so that doesn't change.
Also, for our use-case (leaving), I don't think there will be any future use-case where a leaving unstaked validators would be set with a different shard ID.

Reply:
Ok